File: backend/src/main/java/com/backend/bikerental/entity/VehicleImage.java
package com.backend.bikerental.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "vehicle_images")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class VehicleImage {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(length = 36)
    private String id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "vehicle_id", nullable = false)
    private Vehicle vehicle;

    @Column(name = "image_url", nullable = false, length = 500)
    private String imageUrl;

    @Column(name = "alt_text", length = 255)
    private String altText;

    @Column(name = "display_order")
    private Integer displayOrder = 0;

    @Column(name = "is_primary")
    private Boolean isPrimary = false;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    void prePersist() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
        if (displayOrder == null) {
            displayOrder = 0;
        }
        if (isPrimary == null) {
            isPrimary = false;
        }
    }
}

File: frontend/packages/types/src/vehicleImage.ts
export interface VehicleImage {
  id: string;
  vehicleId: string;

  imageUrl: string;

  altText?: string;

  displayOrder: number;

  isPrimary: boolean;

  createdAt: string;
}

export interface VehicleImageCreatePayload {
  file: File;

  altText?: string;

  displayOrder?: number;

  isPrimary?: boolean;
}

export interface VehicleImageUpdatePayload {
  file?: File;

  altText?: string;

  displayOrder?: number;

  isPrimary?: boolean;
}
